src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -447,12 +451,27 @@ trait Implicits {
         reporter.echo(pos, s"$what is not a valid implicit value for $pt because:\n$reason")
       SearchFailure
     }
+
+    /** Check a case when `info1` is `ArrayCharSequence` and `info2` is `charArrayOps` from `scala.Predef`
+     *
+     * It can be true only when both this methods are imlicit the same type
+     * and it is a case when both are implemented `isEmpty`
+     * because `java.lang.CharSequence` has `isEmpty` since JDK 15
+     *
+     * Force to select scala implementation of `isEmpty` from `charArrayOps`
+     * because it produces bytecode that can be run on JDK before 15
+     */
+    def resolveJDK15CharSequenceConflict(info1: ImplicitInfo, info2: ImplicitInfo) =
```

Comment:
can we also limit this to the situation when we are searching for a conversion to `?{def isEmpty}`? This code lives within the class `class ImplicitSearch(val tree: Tree, val pt: Type, val isView: Boolean, val context0: Context, val pos0: Position = NoPosition, val isByNamePt: Boolean = false)`, so there's quite a bit of context around.